### ISSUE TITLE
Segmentation fault when writing an armadillo function

### ISSUE THREAD

**Issue by dangulod** [RcppCore/RcppArmadillo#328]
Hi,

I am trying to make a function with `RcppArmadillo 0.10.2.2.0` but I am getting an error. The function is:

```
#include <RcppArmadillo.h>

// [[Rcpp::depends(RcppArmadillo)]]
using namespace Rcpp;

// [[Rcpp::export]]
arma::mat test(arma::mat cor)
{
  return cor;
}
```

But when I use `test` function, I get the following error

```
test(x)

 *** caught segfault ***
address 0x7ffe00000000, cause 'memory not mapped'

Traceback:
 1: test(x)

Possible actions:
1: abort (with core dump, if enabled)
2: normal R exit
3: exit R without saving workspace
4: exit R saving workspace

```

I do not know what is wrong, because I have used previous versions of `RcppArmadillo` like this, and I did not get this error. Thanks

**Comment by eddelbuettel** [comment]
First, let's try to avoid identifiers that may clash so `test` and `cor` are tricky.

Second, "weird segfaults" sometimes stem from compiler or library changes. You didn't switch anything between compiling `RcppArmadillo` and `Rcpp`?  Can you reinstall both and try?

Lastly, your example is not complete as there is no input so I made something up.  Full log follows. R 4.0.4, Ubuntu 20.10.

```sh
edd@rob:~$ R

R version 4.0.4 (2021-02-15) -- "Lost Library Book"
Copyright (C) 2021 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.


Happy families are all alike; every unhappy family is unhappy in its own way.
Leo Tolstoy

and every messy data is messy in its own way - it's easy to define the characteristics of a clean dataset (rows are observations, columns are variables, columns contain values of consistent types).
If you start to look at real life data you'll see every way you can imagine data being messy (and many that you can't)!
   -- Hadley Wickham (answering 'in what way messy data sets are messy')
      R-help (January 2008)

> packageVersion("RcppArmadillo")
[1] ‘0.10.2.2.0’
> Rcpp::cppFunction("arma::mat foo(arma::mat x) { return x + x; }", depends="RcppArmadillo")
> foo(matrix(1:4,2,2))
     [,1] [,2]
[1,]    2    6
[2,]    4    8
> 
```

**Comment by eddelbuettel** [comment]
And for completeness you function (unaltered) also sources fine:

```r
> Rcpp::sourceCpp("/tmp/issue328.cpp")
> test(matrix(1:4,2,2))
     [,1] [,2]
[1,]    1    3
[2,]    2    4
> 
```

So I suggest we close this. I cannot see a relevant issue here.

**Comment by dangulod** [comment]
Thank you @eddelbuettel. You are right, it is not a library problem. I am writing a wrapper of a C++ library that uses armadillo so I think there is miss compatibility between them (The system and the RcppArmadillo)

**Comment by coatless** [comment]
I thought that the system link issue was fixed ala @conradsnicta in #281 (addressing #280 and #264) by setting `#define ARMA_DONT_USE_WRAPPER` in the forward declaration?
